st2common/tests/unit/test_config_loader.py

```diff
@@ -518,6 +518,68 @@ def test_get_config_dynamic_config_item_nested_list(self):
 
         config_db.delete()
 
+    def test_get_config_dynamic_config_item_under_additional_properties(self):
+        pack_name = "dummy_pack_schema_with_additional_properties_1"
+        loader = ContentPackConfigLoader(pack_name=pack_name)
+
+        KeyValuePair.add_or_update(KeyValuePairDB(name="k0", value="v0"))
+        KeyValuePair.add_or_update(
+            KeyValuePairDB(name="k1_encrypted", value="v1_encrypted", secret=True)
+        )
+
+        ####################
+        # values in objects under an object with additionalProperties
+        values = {
+            "profiles": {
+                "dev": {
+                    # no host or port to test default value
+                    "token": "hard-coded-secret",
+                },
+                "stage": {
+                    "host": "127.0.0.1",
+                    "port": 8181,
+                    # unencrypted in datastore
+                    "token": "{{st2kv.system.k0}}",
+                },
+                "prod": {
+                    "host": "127.1.2.7",
+                    "port": 8282,
+                    # encrypted in datastore
+                    "token": "{{st2kv.system.k1_encrypted}}",
```

Comment:
If user wants that value in the config to actually be decrypted, they still need to use ``decrypt_kv`` Jinja filter, right?

Reply:
No. Config is automatically decrypted when the schema lists the key as `secret: true`

Reply:
See `_get_datastore_value_for_expression()` in `config_loader.py`

Reply:
Thanks for clarification - I was just somewhat confused by the test case aka something didn't add up.

Reply:
I pushed a fix for this test - 574034f7636350dbb3ed81c3489278fec3b38fe6.

That's the part which confused me a bit.

I assumed we don't do decryption because the value which was stored and the one which was retrieved was supposed to be the same (but the KVP model layer doesn't perform any encryption and we need to encrypt the value ourselves when storing it).

Reply:
Ah! That's why it seemed to be working before, but wasn't. Thanks for the fix!